src/domain/vocabulary/helper/dependencies.ts

```diff
@@ -0,0 +1,148 @@
+    limit
+  } as const)
+
+/**
+ *  Error when withLimit() function is called with a negative limit argument
+ */
+export type NegativeLimitError = ReturnType<typeof NegativeLimitError>
+
+export const MandatoryDependenciesError = (deps: TransientDeps) =>
+  ({
+    _tag: 'mandatory-dependencies',
+    deps
+  } as const)
+
+/**
+ *  Error when one oor more properties are missing in the created Dependencies
+ */
+export type MandatoryDependenciesError = ReturnType<typeof MandatoryDependenciesError>
+
+export type VocabularyDependenciesError =
+  | LanguageEmptyError
+  | NegativeLimitError
+  | MandatoryDependenciesError
+
+export const ShowVocabularyDependenciesError: Show<VocabularyDependenciesError> = {
+  show: (error: VocabularyDependenciesError): string => {
+    switch (error._tag) {
+      case 'language-empty': {
+        return `Error ${error._tag}: The given language parameter cannot be empty`
+      }
+      case 'negative-limit': {
+        return `Error ${error._tag}: The given limit parameter <${error.limit}> cannot be negative`
+      }
+      case 'mandatory-dependencies': {
+        return `Error ${
+          error._tag
+        }: All properties are mandatory, but received a partial object: <${JSON.stringify(
+          error.deps
+        )}>`
+      }
+    }
+  }
+}
+
+const depsIsRequiredTransientDeps = (deps: TransientDeps): deps is Required<TransientDeps> =>
+  'vocabularyGateway' in deps && 'limit' in deps && 'language' in deps
+
+export const withVocabularyGateway =
+  (vocabularyGateway: VocabularyPort): Option =>
+  (deps: TransientDeps) =>
+    E.of({
+      ...deps,
+      vocabularyGateway
+    })
+
+export const withLanguage =
+  (lng: string): Option =>
+  (deps: TransientDeps) =>
+    pipe(
+      lng,
+      E.fromPredicate(P.not(S.isEmpty), LanguageEmptyError),
+      E.map(language => ({
+        ...deps,
+        language
+      }))
+    )
+
+export const withLimit =
+  (limit: number): Option =>
+  (deps: TransientDeps) =>
+    pipe(
+      limit,
+      E.fromPredicate(x => ORD.gt(N.Ord)(x, 0), NegativeLimitError),
+      E.map(l => ({
+        ...deps,
+        limit: l
+      }))
+    )
+
+const validateDeps = (
+  deps: TransientDeps
+): E.Either<VocabularyDependenciesError, RetrieveVocabularyDependencies> =>
+  pipe(
+    deps,
+    O.fromPredicate(depsIsRequiredTransientDeps),
+    O.map(({ vocabularyGateway, limit, language }) => ({
+      _tag: retrieveVocabularyDependenciesTag as typeof retrieveVocabularyDependenciesTag,
+      vocabularyGateway,
+      limit,
+      language
+    })),
+
+    E.fromOption(() => MandatoryDependenciesError(deps))
+  )
+
+export const createVocabularyDependenciesWithOptions = (
+  ...options: Option[]
+): E.Either<VocabularyDependenciesError, RetrieveVocabularyDependencies> =>
+  pipe(
+    options.reduce(
+      (acc: E.Either<VocabularyDependenciesError, TransientDeps>, cur: Option) =>
+        pipe(acc, E.chain(cur)),
+      E.right({})
+    ),
```

Comment:
We can use the `reduce` utility from ReadonlyArray module:

```suggestion
    options,
    RA.reduce(
      E.right({}),
      (acc: E.Either<VocabularyDependenciesError, TransientDeps>, cur: Option) =>
        pipe(acc, E.chain(cur))
    ),
```